figures/views.py

```diff
@@ -420,14 +420,29 @@ class LearnerMetricsViewSet(CommonAuthMixin, viewsets.ReadOnlyModelViewSet):
     search_fields = ['username', 'email', 'profile__name']
     ordering_fields = ['username', 'email', 'profile__name', 'is_active', 'date_joined']
 
+    def query_param_course_keys(self):
+        """
+        TODO: Mixin this
+        """
+        cid_list = self.request.GET.getlist('course')
+        return [CourseKey.from_string(elem.replace(' ', '+')) for elem in cid_list]
```

Comment:
The `.replace(' ', '+')` is an anti pattern in the backend and is error-prune. The frontend should send properly escaped course keys with `encodeURIComponent` or use `fetch()` to take care of that.

```suggestion
    def query_param_course_keys(self):
        """
        TODO: Mixin this
        """
        cid_list = self.request.GET.getlist('course')
        return [CourseKey.from_string(elem) for elem in cid_list]
```

Reply:
I"ll make this change right after I have some coffee :)

Reply:
Well, I would love to make this change, but when I try it, I get this. So the anti-pattern appears to be required still.

```
>>> cid = 'course-v1:StarFleetAcademy SFA01 2161'
>>> from opaque_keys.edx.keys import CourseKey
>>> CourseKey.from_string(cid)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/jbaldwin/.virtualenvs/figtahoe2/lib/python2.7/site-packages/opaque_keys/__init__.py", line 197, in from_string
    return cls.deprecated_fallback._from_deprecated_string(serialized)
  File "/Users/jbaldwin/.virtualenvs/figtahoe2/lib/python2.7/site-packages/opaque_keys/edx/locator.py", line 379, in _from_deprecated_string
    raise InvalidKeyError(cls, serialized)
InvalidKeyError: <class 'opaque_keys.edx.locator.CourseLocator'>: course-v1:StarFleetAcademy SFA01 2161
```

Reply:
@johnbaldwin that's right, `course-v1:StarFleetAcademy SFA01 2161` is _not_ a valid key. Therefore the frontend should not send unescaped which what would make it an invalid key.

It's as if the frontend keeps adding a 999 to numbers and asking the API to subtract 999 before using the `page` argument. This will be more of an issue in the cases like:

 - One of our customers tries to use the figures API through a library that escapes the parameters properly like Python `requests` or pretty much every HTTP client out there.
 - Or if it's us working with Figures APIs through such libraries.

It's bothering that we're doing an extra step every time we make a Figures API to fix something in the _wrong_ place.